ingestion/src/metadata/ingestion/source/mysql.py

```diff
@@ -19,44 +19,45 @@
 from metadata.generated.schema.metadataIngestion.workflow import (
     OpenMetadataServerConfig,
 )
+from metadata.generated.schema.metadataIngestion.workflow import (
+    Source as MetadataIngestionSourceConfig,
+)
+from metadata.generated.schema.metadataIngestion.workflow import SourceConfig
 from metadata.ingestion.api.common import Entity
 from metadata.ingestion.source.sql_source import SQLSource
-from metadata.ingestion.source.sql_source_common import SQLConnectionConfig
-
-
-class MySQLConfig(MysqlConnection, SQLConnectionConfig):
-    def get_connection_url(self):
-        return super().get_connection_url()
 
 
 class MysqlSource(SQLSource):
     def __init__(self, config, metadata_config):
         super().__init__(config, metadata_config)
 
     @classmethod
-    def create(cls, config_dict, metadata_config: OpenMetadataServerConfig):
-        config = MySQLConfig.parse_obj(config_dict)
+    def create(cls, config_dict, metadata_config_dict):
+        config = MetadataIngestionSourceConfig.parse_obj(config_dict)
+        metadata_config = OpenMetadataServerConfig.parse_obj(metadata_config_dict)
+        # Added parsing of MysqlConnection in order to validate config with pydantic models
+        # Missing / Extra fields will throw errors
+        MysqlConnection.parse_obj(config_dict.get("serviceConnection").get("config"))
+        SourceConfig.parse_obj(config_dict.get("sourceConfig"))
```

Comment:
We should not need to recast `SourceConfig` here. Any validation error will be picked up at `config = MetadataIngestionSourceConfig.parse_obj(config_dict)`. For the connection, let's just run a `isinstance` and call it a day. With the schemas we can have great docs to set this up. If the overall structure matches, showing an error if the class is incorrect should be enough